### PR TITLE
PHP CS Fix

### PR DESCRIPTION
**Pull request by charescape**

### CODE REVIEW

**Comment by charescape** [comment]
This PR related to https://github.com/spiral/framework/issues/215

**Comment by wolfy-j** [comment]
Hi, can I get your cs settings or script? There are few corrections I have to make.

**Comment by talentant** [comment]
sure!

```php
<?php

declare(strict_types=1);

return PhpCsFixer\Config::create()
    ->setCacheFile(__DIR__ . '/php_cs.cache')
    ->setRiskyAllowed(true)
    ->setRules([
        '@Symfony' => true,
        '@Symfony:risky' => true,
        '@PHP71Migration' => true,
        '@PHP71Migration:risky' => true,
        '@PHPUnit60Migration:risky' => true,
        'trailing_comma_in_multiline_array' => false,
        'native_constant_invocation' => false,
        'native_function_invocation' => false,
        'protected_to_private' => false,
        'no_alias_functions' => false,
        'set_type_to_cast' => false,
        'phpdoc_inline_tag' => false,
        'function_to_constant' => false,
        'self_accessor' => false,
        'increment_style' => false,
        'phpdoc_var_without_name' => false,
        'binary_operator_spaces' => [
            'default' => 'align',
        ],
        'array_syntax' => [
            'syntax' => 'short',
        ],
        'is_null' => false,
        'cast_spaces' => [
            'space' => 'none',
        ],
        'declare_strict_types' => true,
        'ordered_imports' => [
            'sortAlgorithm' => 'alpha',
            'importsOrder' => [
                'const',
                'function',
                'class',
            ],
        ],
        'no_useless_else' => true,
        'no_useless_return' => true,
        'php_unit_construct' => true,
        'php_unit_strict' => true,
        'phpdoc_separation' => false,
        'concat_space' => [
            'spacing' => 'one'
        ],
        'phpdoc_align' => [
            'tags' => []
        ],
        'phpdoc_annotation_without_dot' => true,
        'phpdoc_no_alias_tag' => false,
        'yoda_style' => false,
        'phpdoc_to_comment' => false,
    ])
    ->setFinder(
        PhpCsFixer\Finder::create()
            ->in(__DIR__)
            ->exclude([
                'vendor',
                'bin',
                'tests',
            ])
            ->notPath('functions.php')
    );

```

**Comment by charescape** [comment]
```php
<?php

declare(strict_types=1);

return PhpCsFixer\Config::create()
    ->setCacheFile(__DIR__ . '/php_cs.cache')
    ->setRiskyAllowed(true)
    ->setRules([
        '@Symfony' => true,
        '@Symfony:risky' => true,
        '@PHP71Migration' => true,
        '@PHP71Migration:risky' => true,
        '@PHPUnit60Migration:risky' => true,
        'trailing_comma_in_multiline_array' => false,
        'native_constant_invocation' => false,
        'native_function_invocation' => false,
        'protected_to_private' => false,
        'no_alias_functions' => false,
        'set_type_to_cast' => false,
        'phpdoc_inline_tag' => false,
        'function_to_constant' => false,
        'self_accessor' => false,
        'increment_style' => false,
        'phpdoc_var_without_name' => false,
        'binary_operator_spaces' => [
            'default' => 'align',
        ],
        'array_syntax' => [
            'syntax' => 'short',
        ],
        'is_null' => false,
        'cast_spaces' => [
            'space' => 'none',
        ],
        'declare_strict_types' => true,
        'ordered_imports' => [
            'sortAlgorithm' => 'alpha',
            'importsOrder' => [
                'const',
                'function',
                'class',
            ],
        ],
        'no_useless_else' => true,
        'no_useless_return' => true,
        'php_unit_construct' => true,
        'php_unit_strict' => true,
        'phpdoc_separation' => false,
        'concat_space' => [
            'spacing' => 'one'
        ],
        'phpdoc_align' => [
            'tags' => []
        ],
        'phpdoc_annotation_without_dot' => true,
        'phpdoc_no_alias_tag' => false,
        'yoda_style' => false,
        'phpdoc_to_comment' => false,
    ])
    ->setFinder(
        PhpCsFixer\Finder::create()
            ->in(__DIR__)
            ->exclude([
                'vendor',
                'bin',
                'tests',
            ])
            ->notPath('functions.php')
    );
```

**Comment by charescape** [comment]
@wolfy-j I followed https://cs.symfony.com/ and created `.php_cs` file, then run `php php-cs-fixer-v2.phar fix`

**Comment by charescape** [comment]
To avoid changing too much code style, I disabled some rules, although I think they make sense.

**Comment by wolfy-j** [comment]
Thank you, considering the impact and 60+ repos we have to apply to I can't guarantee quick merge. I need to talk to all the teams to make sure that CS is aligned.

**Comment by wolfy-j** [comment]
FYI, we did not forget about it, @alexndr-novikov is working on the full-size analysis of the best approach.

**Comment by talentant** [comment]
Thank you for your excellent work. 👍

**Comment by alexndr-novikov** [comment]
Thank you @charescape , we highly appreciate your contribution.

Unfortunately, this PR requires some modifications to be merged. It does not fit PSR-12, and has some changes that are not required by it, but do some unnecessary code changes. (like php-cs-fix 'binary_operator_spaces' alignment with '=' sign and etc.)
Also it has conflicts with current codebase.

These's one problem : php-cs-fixer does not support PSR-12 yet (it is expected in v3, https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4502), so now you'll need 2 tools to fix the code in proper way and check it: code sniffer that already supports PSR-12 checks, and PHP-CS-Fixer.

So, please follow the instruction to update this PR (or optionally close this one, re-fork and apply from scratch and then create new one)

1. Add to composer.json libraries:
```
composer require --dev  friendsofphp/php-cs-fixer:^v2.15.3
composer require --dev squizlabs/php_codesniffer:^3.5.0
```
2. add .php_cs file to the root of the repo with the following content: (and attach it to PR later)

```php
<?php

declare(strict_types=1);

return PhpCsFixer\Config::create()
    ->setCacheFile(__DIR__ . '/.php_cs.cache')
    ->setRiskyAllowed(true)
    ->setRules([
        '@PSR2' => true,
        'binary_operator_spaces' => [
            'default' => null,
            'operators' => [
                '|' => 'single_space',
                '!==' => 'single_space',
                '!=' => 'single_space',
                '==' => 'single_space',
                '===' => 'single_space',
            ]
        ],
        'ordered_class_elements' => true,
        'trailing_comma_in_multiline_array' => false,
        'declare_strict_types' => true,
        'linebreak_after_opening_tag' => true,
        'blank_line_after_opening_tag' => true,
        'single_quote' => true,
        'lowercase_cast' => true,
        'short_scalar_cast' => true,
        'no_leading_import_slash' => true,
        'declare_equal_normalize' => [
            'space' => 'none'
        ],
        'new_with_braces' => true,
        'no_blank_lines_after_phpdoc' => true,
        'single_blank_line_before_namespace' => true,
        'visibility_required' => ['property', 'method', 'const'],
        'ternary_operator_spaces' => true,
        'unary_operator_spaces' => true,
        'return_type_declaration' => true,
        'concat_space' => [
            'spacing' => 'one'
        ],
        'no_useless_else' => true,
        'no_useless_return' => true,
        'phpdoc_separation' => false,
        'yoda_style' => false,
        'void_return' => true,
    ])
    ->setFinder(
        PhpCsFixer\Finder::create()
            ->in(__DIR__)
            ->exclude([
                'vendor',
                'bin',
            ])
    );
```


3. Run
```
./vendor/bin/php-cs-fixer fix
./vendor/bin/phpcbf --report=code --standard=PSR12 --colors tests src
./vendor/bin/phpcs -n --report=code --standard=PSR12 --colors tests src
```

If last check still alert any issues (usually about class docblock and declare order and blank lines) - fix it manually please

4. Add to .travis.yml on line 30

vendor/bin/phpcs -n --report=code --standard=PSR12 --colors tests src

And then update the pull request.

Let me know if you have any questions about it

**Comment by charescape** [comment]
@alexndr-novikov Thank you very much. I am going to follow your instruction and update this PR soon.

**Comment by charescape** [comment]
> 1. Add to composer.json libraries

done.

> 2. add .php_cs file to the root of the repo with the following content: (and attach it to PR later)

done.

> 3. Run auto fix

done.

> 4. Do manually fix

done.

> 5. Add to .travis.yml on line 30

done.

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ install:
 
 script:
   - vendor/bin/phpunit --coverage-clover=coverage.xml
+  - vendor/bin/phpcs -n --report=code --standard=PSR12 --colors tests src
   - vendor/bin/phpcs -n --standard=PSR2 --colors src tests
```
Comment (alexndr-novikov) [inline .travis.yml]: remove line 31 please, `- vendor/bin/phpcs -n --standard=PSR2 --colors src tests` it became duplicate with less functionality with the new added line

Comment (charescape) [reply .travis.yml]: removed.

**Comment by wolfy-j** [comment]
@charescape if you want you can use spiral/code-style by youself, so the PR will remain after you.

**Comment by wolfy-j** [comment]
@charescape `vendon/bin/spiral-cs fix src tests functions.php`

**Comment by talentant** [comment]
@wolfy-j Thank you! I will try to update this PR today.

**Comment by charescape** [comment]
@wolfy-j Thank you! I will try to update this PR today.

**Comment by wolfy-j** [comment]
I'll push an update with enabled CS to it. Travis will fail for now, so we will see how the CS will work.

**Comment by alexndr-novikov** [comment]
@charescape 
so the flow should be like
1. composer require --dev spiral/code-style
2. ./vendor/bin/spiral-cs fix functions.php src/ tests
3. ./vendor/bin/spiral-cs check functions.php src/ tests
4. fix the rest issues manually (declare order, ini_set('display_errors', '1');, etc)

**Comment by charescape** [comment]
@alexndr-novikov Thanks for your instruction. I will do it soon.

**Comment by wolfy-j** [comment]
Hi, we are prepping big release, when do you think we can merge it? :|

**Comment by charescape** [comment]
@wolfy-j @alexndr-novikov Sorry, I'm still busy at the moment, and I don't want to delay the progress of your project, so this PR should be closed.

**Comment by wolfy-j** [comment]
Sorry about that.

**Pull request closed without merging**